File: ProiectColectiv/ProiectColectiv.Application/Models/SubjectModel.cs
using ProiectColectiv.Domain.Entities;
using ProiectColectiv.Domain.Enums;

namespace ProiectColectiv.Application.Models;

public class SubjectModel
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public int Credits { get; set; }
    public MajorEnum Major { get; set; }
    public SubjectTypeEnum SubjectType { get; set; }
}

File: proiect-colectiv-frontend/src/app/models/subjects/SubjectModel.ts
import { BaseModel } from '../base/BaseModel';
import { MajorEnum } from '../enums/MajorEnums';
import { SubjectTypeEnum } from '../enums/SubjectEnums';

export class SubjectModel extends BaseModel {
  name: string;
  credits: number;
  major: MajorEnum;
  subjectType: SubjectTypeEnum;

  constructor(
    id: string,
    name: string,
    credits: number,
    major: MajorEnum,
    subjectType: SubjectTypeEnum
  ) {
    super(id);
    this.name = name;
    this.credits = credits;
    this.major = major;
    this.subjectType = subjectType;
  }
}
